format.go

```diff
@@ -64,6 +69,52 @@ func ParseISO(value string) (Date, error) {
 	return Date{encode(t)}, nil
 }
 
+func parseError(input string) error {
+	return fmt.Errorf("Date.ParseISO: length parse %s", input)
+}
+
+func parseISORune(value string) (Date, error) {
+	if len(value) < 10 {
+		return Date{}, parseError(value)
+	}
+
+	data := []int{0, 0, 0}
+	yearsign := 1
+	state := 0
+	characters := 0
+	for i, c := range value {
+		if i == 0 && (c == '+' || c == '-') {
+			if c == '-' {
+				yearsign = -1
+			}
+			continue
+		}
+
+		if state >= 3 {
+			if c != 'T' {
+				return Date{}, parseError(value)
+			}
+			break
+		}
+
+		if unicode.IsDigit(c) {
+			data[state] = data[state]*10 + int(c-'0')
```

Comment:
Is that not abusing the fact that runes are `int32`?

Reply:
not at all.